DeskReserve/Controllers/FloorsController.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using DeskReserve.Data.DBContext.Entity;
+using DeskReserve.Domain;
+using Microsoft.AspNetCore.JsonPatch;
+
+namespace DeskReserve.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FloorsController : ControllerBase
+    {
+        private readonly FloorService _floorService;
+
+        public FloorsController(FloorService floorService)
+        {
+            _floorService = floorService;
+        }
+
+        // GET: api/Floors
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Floor>>> GetFloors()
+        {
+            var floors = await _floorService.GetFloors();
+            return Ok(floors);
+        }
+
+        // GET: api/Floors/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Floor>> GetFloor(Guid id)
+        {
+            var floor = await _floorService.GetFloor(id);
+
+            if (floor == null)
+            {
+                return NotFound();
+            }
+
+            return floor;
+        }
+
+        // PUT: api/Floors/5
+        [HttpPut("{id}")]
+        public async Task<IActionResult> PutFloor(Guid id, Floor floor)
+        {
+            if (id != floor.FloorId)
+            {
+                return BadRequest();
+            }
+
+            var success = await _floorService.UpdateFloor(id, floor);
+
+            return success ? NoContent() : NotFound();
+        }
+
+        // POST: api/Floors
+        [HttpPost]
+        public async Task<ActionResult<Floor>> PostFloor(Floor floor)
+        {
+            var newFloor = await _floorService.CreateFloor(floor);
+            return CreatedAtAction("GetFloor", new { id = newFloor.FloorId }, newFloor);
```

Comment:
Get flor is the wrong action here

Reply:
Disagree. CreatedAtAction generates a "201: Created" response with a Location header that specifies the URI of the newly created resource. The endpoint of which the new resource will be accessed will be [BaseUri]/api/Floors/{id} which is the GetFloor action we have in the controller.

Reply:
I see, no need to return response header at what action you can access it, because its not the only one, If we remove the Get logic the Create logic is misleading. That is not robust software. Also with what "action " are you going to replace it if we remove the get logic?